client/checkout/api/index.js

```diff
@@ -344,12 +344,12 @@ export default class WCPayAPI {
 		} )
 			.then( ( response ) => {
 				if ( 'failure' === response.result ) {
-					throw response.messages;
+					throw { message: response.messages };
 				}
 				return response;
 			} )
 			.catch( ( error ) => {
-				throw `Error submitting form! ${ error.status }: ${ error.statusText }.`;
+				throw error;
```

Comment:
```suggestion
				if( error.message ) {
				    throw error;
				} else {
				    throw new Error( `Error submitting form! ${ error.status }: ${ error.statusText }.` );
				}
```
Currently, just throwing the error here is a little redundant, so we don't really need to have this catch block. 

However, @superdav42 noticed that if this AJAX request times out or returns a `500` for some reason, then the `request` function rejects and the error thrown is a [jqxhr object](https://api.jquery.com/jQuery.ajax/#jqXHR) instead of the error object we would expect from `wp_send_json_error`. You can test this by using the debugger to cause a timeout or simply putting some broken code inside the `process_payment` function that will force the response to be a `500` error. 

I implemented that previous `Error submitting form!` message, using the `error.status` and `error.statusText` properties to handle this case. You're welcome to come up with a more user friendly error message than the one I used though! I think this case is necessary though, because otherwise a server error here just leaves JS errors in the console and doesn't tell the user that something has gone wrong.

I think similar logic to this should also probably be implemented around the `createIntent` function as well.

Reply:
I have added the case, thanks for pointing it out!